torch_brain/utils/stitcher.py

```diff
@@ -376,11 +357,38 @@ def on_validation_epoch_end(self, trainer, pl_module, prefix="val"):
                         {f"{prefix}_metrics": wandb.Table(dataframe=metrics_df)}
                     )
 
-    def on_test_epoch_start(self, *args, **kwargs):
-        self.on_validation_epoch_start(*args, **kwargs)
+    def on_test_epoch_start(self, trainer, pl_module):
+        self._on_epoch_start_helper(trainer, mode="test")
 
     def on_test_batch_end(self, *args, **kwargs):
         self.on_validation_batch_end(*args, **kwargs)
 
     def on_test_epoch_end(self, *args, **kwargs):
         self.on_validation_epoch_end(*args, **kwargs, prefix="test")
+
+    def _on_epoch_start_helper(self, trainer, mode: str = "val"):
```

Comment:
Nit: Can we have a more explicit name for this function? Maybe `_setup_caches(...)`?